Mathlib/Combinatorics/SimpleGraph/Connectivity.lean

```diff
@@ -2577,6 +2577,29 @@ theorem Walk.mem_finsetWalkLength_iff_length_eq {n : ℕ} {u v : V} (p : G.Walk
 
 variable (G)
 
+/-- The `Finset` of walks from `u` to `v` with length less than `n`. See `finsetWalkLength`. --/
+def finsetWalkLengthLt (n : ℕ) (u v : V) : Finset (G.Walk u v) :=
+  (Finset.range n).disjiUnion
+    (fun l ↦ G.finsetWalkLength l u v)
+    (fun l _ l' _ hne _ hsl hsl' p hp ↦
+      have hl : p.length = l := p.mem_finsetWalkLength_iff_length_eq.mp (hsl hp)
+      have hl' : p.length = l' := p.mem_finsetWalkLength_iff_length_eq.mp (hsl' hp)
+      False.elim <| hne <| hl.symm.trans hl')
+
+open Finset in
+theorem coe_finsetWalkLengthLt_eq (n : ℕ) (u v : V) :
+    (G.finsetWalkLengthLt n u v : Set (G.Walk u v)) = {p : G.Walk u v | p.length < n} := by
+  ext p
+  simp [finsetWalkLengthLt, mem_coe, mem_disjiUnion, Walk.mem_finsetWalkLength_iff_length_eq]
+
+variable {G}
+
+theorem Walk.mem_finsetWalkLengthLt_iff_length_lt {n : ℕ} {u v : V} (p : G.Walk u v) :
```

Comment:
Iffs take implicit arguments
```suggestion
theorem mem_finsetWalkLengthLT {n : ℕ} {u v : V} {p : G.Walk u v} :
```

Reply:
Ah, I just adapted the lemma from https://github.com/leanprover-community/mathlib4/blob/bc3ab3eddef1b92b87d321f25c74899502c25239/Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean#L112 - should we change it there too then?